pkg/sidecarexec/sandbox_cmd_runner_linux.go

```diff
@@ -0,0 +1,282 @@
+		if !strings.HasPrefix(path, limitDir) {
+			return fmt.Errorf("Expected visible path to start with '%s', but was '%s'", limitDir, path)
+		}
+		mounts = append(mounts, mountPoint{Path: path[1:], Writeable: true})
+	}
+
+	if args.Posix {
+		// Example: calling out to git
+		mounts = append(mounts, []mountPoint{
+			{Path: "etc", NonMount: true},
+			{Path: "etc/passwd", File: true, NonMount: true, Content: "root:x:0:0:/home/kapp-controller:/usr/sbin/nologin\n"},
+			{Path: "bin"}, // e.g. bin/sh was referenced by git executables
+			{Path: "lib"},
+			{Path: "lib64"}, // some dynamic libraries
+			{Path: "var/lib"},
+			{Path: "usr/lib"},
+			{Path: "usr/libexec"}, // internal git binaries
+			{Path: "usr/share"},   // internal git data
+			{Path: "usr/sbin"},
+			// TODO get rid of 'dev/null'? (used by vendir today)
+			{Path: "dev", NonMount: true},
+			{Path: "dev/null", NonMount: true, File: true},
+		}...)
+	}
+
+	if args.Network {
+		mounts = append(mounts, []mountPoint{
+			// Cannot mount whole 'etc' directory directly (mount returns exit status 32)
+			// Mounting individual child files/directories makes it work.
+			{Path: "etc/pki"},
+			{Path: "etc/resolv.conf", File: true},
+			{Path: "etc/hostname", File: true},
+			{Path: "etc/hosts", File: true},
+		}...)
+	}
+
+	err := w.configureMounts(mounts, args.NewRootDir)
+	if err != nil {
+		return err
+	}
+
+	err = w.pivotRoot(args.NewRootDir)
+	if err != nil {
+		return err
+	}
+
+	// Not all command specify execution directory
+	if len(args.CmdDir) > 0 {
+		err = os.Chdir(args.CmdDir)
+		if err != nil {
+			return fmt.Errorf("Chdir '%s': %s", args.CmdDir, err)
+		}
+	}
+
+	return syscall.Exec(args.CmdPath, append([]string{args.CmdPath}, args.CmdArgs...), os.Environ())
+}
+
+func (SandboxWrap) configureMounts(mounts []mountPoint, newRootDir string) error {
+	for _, mount := range mounts {
+		if mount.File {
+			file, err := os.Create(filepath.Join(newRootDir, mount.Path))
+			if err != nil {
+				return fmt.Errorf("Creating mount file '%s': %s", mount.Path, err)
+			}
+			if len(mount.Content) > 0 {
+				_, err = file.Write([]byte(mount.Content))
+				if err != nil {
+					return fmt.Errorf("Writing into file '%s': %s", mount.Path, err)
+				}
+			}
+			err = file.Close()
+			if err != nil {
+				return fmt.Errorf("Closing mount file '%s': %s", mount.Path, err)
+			}
+		} else {
+			err := os.MkdirAll(filepath.Join(newRootDir, mount.Path), 0700)
+			if err != nil {
+				return fmt.Errorf("Creating mount dir '%s': %s", mount.Path, err)
+			}
+		}
+	}
+
+	for _, mount := range mounts {
+		if mount.NonMount {
+			continue
+		}
+
+		srcPath := string(filepath.Separator) + mount.Path
+		dstPath := filepath.Join(newRootDir, mount.Path)
+
+		// Equivalent to: mount --bind /usr/sbin usr/sbin
+		err := syscall.Mount(srcPath, dstPath, "bind", syscall.MS_BIND, "")
+		if err != nil {
+			return fmt.Errorf("Mounting path '%s': %s", mount.Path, err)
+		}
+
+		// TODO readonly for files?
```

Comment:
Seems like this TODO is covered by `syscall.MS_RDONLY` right?

Reply:
conditional below uses !mount.File. i had it working with minikube but got stuck on GKE. need to revisit why it was not happy for mounting files.